doc/build-osx.md

```diff
@@ -17,29 +17,61 @@ When the popup appears, click `Install`.
 
 Then install [Homebrew](https://brew.sh).
 
-## Base build dependencies
-
+## Dependencies
 ```shell
-brew install automake libtool pkg-config libnatpmp sqlite
+brew install automake berkeley-db4 libtool boost miniupnpc pkg-config python qt libevent libnatpmp qrencode sqlite
```

Comment:
we need gmp as well. and even with it installed ./configure doesn't find it

Reply:
Then should we include it or add it once detection has been fixed?

Reply:
Okay, we should add gmp stuff here now

Reply:
Documentation has been updated for macOS and Linux (Fedora and Debian flavours)